apis/Google.Cloud.Logging.Console/Google.Cloud.Logging.Console.Tests/GoogleCloudConsoleFormatterTest.cs

```diff
@@ -152,6 +153,101 @@ public void ConsoleLoggerOptions_OptionChange_IsReloaded()
+    public void Log_NullActivity_TraceInformationIsNotLogged()
+    {
+        // Set the TraceGoogleCloudProjectId.
+        var formatter = CreateFormatter(new GoogleCloudConsoleFormatterOptions { TraceGoogleCloudProjectId = "my-project-id" });
+
+        // Set Activity.Current to null for clarity. (It is already null here as no Activity is created.)
+        Activity.Current = null;
+
+        var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
+        var writer = new StringWriter { NewLine = "\n" };
+
+        formatter.Write(logEntry, scopeProvider: null, writer);
+
+        var expectedJson = "{\"message\":\"test\",\"category\":\"LogCategory\",\"severity\":\"INFO\"}\n";
+        var actualJson = writer.ToString();
+
+        Assert.Equal(expectedJson, actualJson);
+    }
+
+    [Fact]
+    public void Log_DefaultTraceId_TraceInformationIsNotLogged()
+    {
+        // Set the TraceGoogleCloudProjectId.
+        var formatter = CreateFormatter(new GoogleCloudConsoleFormatterOptions { TraceGoogleCloudProjectId = "my-project-id" });
+
+        // Backup the generator, so that it can be reset after the test.
+        // It is null by default but just to keep it easy to understand.
+        var generator = Activity.TraceIdGenerator;
+
+        // Simulate the condition where the generated `TraceId` is equal to the default value of `ActivityTraceId`.
+        Activity.TraceIdGenerator = () => default;
+        using var activity = new Activity("my-activity").Start();
+
+        // Reset the generator, so that other tests aren't impacted.
+        Activity.TraceIdGenerator = generator;
+        
+        var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
+        var writer = new StringWriter { NewLine = "\n" };
+        
+        formatter.Write(logEntry, scopeProvider: null, writer);
+
+        var expectedJson = "{\"message\":\"test\",\"category\":\"LogCategory\",\"severity\":\"INFO\"}\n";
+        var actualJson = writer.ToString();
+
+        Assert.Equal(expectedJson, actualJson);
+    }
+
+    [Fact]
+    public void Log_TraceInformationIsLogged()
+    {
+        // Trace information is logged, only if:
+        // 1. TraceGoogleCloudProjectId is set.
+        // 2. Activity.Current is not null; and
+        // 3. Activity.Current.TraceId does not have the default value.
+
+        // Set the TraceGoogleCloudProjectId.
+        string traceProjectId = "my-project-id";
+        var formatter = CreateFormatter(new GoogleCloudConsoleFormatterOptions { TraceGoogleCloudProjectId = traceProjectId });
+
+        // Start an activity.
+        using var activity = new Activity("my-activity").Start();
+
+        var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
```

Comment:
Can we extract this "Log a simple entry and return its JSON representation" as a separate method? It seems to be in common in all of these tests (and possibly some others).